src/pages/docs/creators/ray/cluster-yaml.md

```diff
@@ -104,4 +108,91 @@ By default, Ray on Golem uses prepackaged VM images including [relatively fresh]
 However, you can use these properties to override the detection and request a specific image. 
 Supported tags are available on [Golem registry](https://registry.golem.network/explore/golem/ray-on-golem).
 
-Please [let us know on `#Ray on Golem` discord channel)](https://chat.golem.network/) if you need an image with any specific content. We will be happy to help you.
+Please [let us know on the `#Ray on Golem` discord channel)](https://chat.golem.network/) if you need an image with any specific content. We will be happy to help you.
+
+## Budget control
+
+### Spending hard limit
+
+Within `provider.parameters` section there is the `total_budget` property.
+It defines the maximum amount of GLMs paid for the whole cluster operations - from `ray up` until `ray down`.
+
+At the moment, when the spending reaches the limit, Ray on Golem will stop spending, effectively terminating the cluster nodes.
+
+```yaml
+provider:
+  parameters:
+
+    # Maximum amount of GLMs that's going to be spent for the whole cluster
+    total_budget: 2
+```
+
+### Avoiding too-expensive providers
+
+You can use `provider.parameters.node_config.per_cpu_budget` section to define the limits on providers' prices.
+Ray on Golem won't work with providers who exceed any of the following price settings.
+
+#### Maximum provider prices
+
+Golem providers charge in three ways. They charge:
+- the initial price at the start of image deployment - we divide it by the number of CPUs to work with price per CPU,
```

Comment:
```suggestion
- the initial price at the start of image deployment - we divide it by the number of CPUs so that the price is representative of the actual computing power that you pay for,
```